frontend/main-app/src/components/FormViewer/FormViewer.tsx

```diff
@@ -39,27 +29,27 @@ export const FormViewer = (): React.JSX.Element => {
 
     const formData = new FormData(event.currentTarget as HTMLFormElement);
     const cleanedFormData = cleanFormData(formData);
-    const validationSchema = generateValidationSchema(formSchema);
+    const validationSchema = generateValidationSchema(form);
     const result = validationSchema.safeParse(cleanedFormData);
 
     if ("error" in result) {
-      setFormErrors(result.error.formErrors.fieldErrors);
+      setFormValidationErrors(result.error.formErrors.fieldErrors);
       setFormAlert("validationError");
     } else {
-      setFormErrors({});
-      postSubmission(formSchema, cleanedFormData, setFormAlert);
+      setFormValidationErrors({});
+      postSubmission(form, cleanedFormData, setFormAlert);
     }
   };
 
   const RenderForm = () => (
     <main className={classes.card}>
       <form onSubmit={handleSubmit}>
         <Heading level={1} size="xlarge">
-          {formSchema?.title}
+          {form?.title}
         </Heading>
-        {formSchema?.components.map((component) => (
+        {form?.components.map((component) => (
           <div key={component.name} className={classes.component}>
-            <FormComponent component={component} error={formErrors[component.name]} />
+            <FormComponent component={component} error={formValidationErrors[component.name]} />
           </div>
         ))}
```

Comment:
Just curious about the `?` for `title` and `components`, are the any scenarios where the form doesnt have a title or any components? Is there validation when building the form that stops these cases?

Reply:
If we don't have the `?/optional chaining operator`, the component will try to render before useEffect and getFormSchema has finished its async operation. That will trigger an error, because the form has not yet been recieved. Since the initial state of the `form` state is null, we will get this error:
![image](https://github.com/ErlingHauan/FormFactory/assets/148075168/972188e5-18ac-479f-96c2-e2acfbf091f9)

Another way to solve this, without the optional chaining operator, would be to set the initial state of the `form` state to a form object with preliminary values.

Reply:
Understood :) You can keep as it is.